scipy/integrate/tests/test_quadpack.py

```diff
@@ -261,6 +262,26 @@ def func(x0, x1):
             return x0 + x1 + 1 + 2
         assert_quad(dblquad(func, 1, 2, 1, 2),6.)
 
+    @pytest.mark.parametrize('kwargs, msg', [
+        ({'epsabs': 0.1},
+         "Passing 'epsabs' explicitly is deprecated and will be removed in"
+         " SciPy 1.12.0. Please pass it via the 'opts' argument instead."
+         " 'epsabs' will be ignored."),
+        ({'epsrel': 0.1},
+         "Passing 'epsrel' explicitly is deprecated and will be removed in"
+         " SciPy 1.12.0. Please pass it via the 'opts' argument instead."
+         " 'epsrel' will be ignored.")])
```

Comment:
This can be simplified considerably if you just parameterize over `kwarg` `epsabs` or `epsrel` and define the message with an f-string.